src/pages/SearchUserPage.tsx

```diff
@@ -0,0 +1,155 @@
+import { FormEvent, Fragment } from "react";
+
+import { useAtom, useAtomValue } from "jotai";
+
+import { css } from "@styled-system/css";
+
+import { IngredientKey } from "@/types/ingredient";
+
+import { Link } from "@/routes/Link";
+import { $getSearchedUsers, $nickname } from "@/store/user";
+import Header from "@/components/common/Header";
+import Input from "@/components/common/Input";
+import Loading from "@/components/common/Loading";
+import { INGREDIENT_ICON_BY_KEY, INGREDIENT_NAME_BY_KEY } from "@/constants/ingredient";
+import SearchIcon from "@/assets/svg/search.svg";
+
+const SearchUserPage = () => {
+  const [nickname, setNickname] = useAtom($nickname);
+  const { data: searchedUsers, isLoading } = useAtomValue($getSearchedUsers(nickname));
+
+  const handleSubmitForm = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const formData = new FormData(event.currentTarget);
+    const nickname = formData.get("nickname") as string;
+
+    setNickname(nickname);
+  };
+
+  return (
+    <Fragment>
+      <Header hasPreviousPage>검색</Header>
+      <div className={styles.container}>
+        <form onSubmit={handleSubmitForm} className={styles.form}>
+          <label htmlFor="nickname" className="a11y-hidden">
+            검색
+          </label>
+          <Input id="nickname" name="nickname" placeholder="찾으시는 분의 닉네임을 입력하세요" />
+          <button className={styles.searchButton}>
+            <SearchIcon />
+          </button>
+        </form>
+        {isLoading && <Loading />}
+        <div>
+          <ul className={styles.searchResult}>
+            {searchedUsers?.length === 0 && (
+              <div className={styles.noUsers}>일치하는 사용자가 없습니다.</div>
+            )}
+            {searchedUsers?.map(({ id, nickname, primaryIngredient }) => (
+              <Ingredient key={id} id={id} nickanme={nickname} ingredientKey={primaryIngredient} />
+            ))}
+          </ul>
+        </div>
```

Comment:
```suggestion
        {!isLoading && <div>
          <ul className={styles.searchResult}>
            {searchedUsers.length === 0 && (
              <div className={styles.noUsers}>일치하는 사용자가 없습니다.</div>
            )}
            {searchedUsers.map(({ id, nickname, primaryIngredient }) => (
              <Ingredient key={id} id={id} nickanme={nickname} ingredientKey={primaryIngredient} />
            ))}
          </ul>
        </div>}
```

혹시 저 !isLoading으로 걸러주면 searchedUsers가 보장되지 않나요?

Reply:
맞습니다! isLoading 분기처리를 해주는 편이 옵셔널을 사용하는 것 보다 조금 더 명시적인 것 같네요